app/queries/location_query.rb

```diff
@@ -8,31 +8,65 @@ class LocationQuery < ApplicationQuery
 
   private
 
-  def call(field_name, location_query, radius_in_miles)
-    normalised_query = location_query&.strip&.downcase
+  def call(field_name, location_query, radius_in_miles, sort_by_distance: false)
+    normalised_query = normalise_query(location_query)
     radius = convert_miles_to_metres(radius_in_miles.to_i)
 
-    if normalised_query.blank? || NATIONWIDE_LOCATIONS.include?(normalised_query)
-      scope
-    elsif LocationPolygon.contain?(normalised_query)
-      polygon = LocationPolygon.with_name(normalised_query)
-
-      scope
-        .joins("
-          INNER JOIN location_polygons
-          ON ST_DWithin(#{field_name}, location_polygons.area, #{radius})
-        ")
-        .where("location_polygons.id = ?", polygon.id)
+    return scope if normalised_query.blank? || nationwide_location?(normalised_query)
+
+    if polygon_location?(normalised_query)
+      handle_polygon_location(field_name, normalised_query, radius, sort_by_distance)
     else
-      coordinates = Geocoding.new(normalised_query).coordinates
+      handle_coordinates(field_name, normalised_query, radius, sort_by_distance)
+    end
+  end
+
+  def normalise_query(query)
+    query&.strip&.downcase
+  end
 
-      # TODO: Geocoding class currently returns this on error, it should probably raise a
-      # suitable error instead. Refactor later!
-      return scope.none if coordinates == [0, 0]
+  def nationwide_location?(query)
+    NATIONWIDE_LOCATIONS.include?(query)
+  end
 
-      point = "POINT(#{coordinates.second} #{coordinates.first})"
+  def polygon_location?(query)
+    LocationPolygon.contain?(query)
+  end
 
-      scope.where("ST_DWithin(#{field_name}, ?, ?)", point, radius)
-    end
+  def handle_polygon_location(field_name, query, radius, sort_by_distance)
+    polygon = LocationPolygon.with_name(query)
+    @scope = scope.joins("
+      INNER JOIN location_polygons
+      ON ST_DWithin(#{field_name}, location_polygons.area, #{radius})
+    ").where("location_polygons.id = ?", polygon.id)
+
+    sort_by_polygon_distance(field_name) if sort_by_distance
+
+    scope
+  end
+
+  def handle_coordinates(field_name, query, radius, sort_by_distance)
+    coordinates = Geocoding.new(query).coordinates
+
+    # TODO: Geocoding class currently returns this on error, it should probably raise a
+    # suitable error instead. Refactor later!
+    return scope.none if coordinates == [0, 0]
+
+    point = "POINT(#{coordinates.second} #{coordinates.first})"
+    @scope = scope.where("ST_DWithin(#{field_name}, ?, ?)", point, radius)
+
+    sort_by_coordinates_distance(field_name, point) if sort_by_distance
+
+    scope
+  end
+
```

Comment:
I have included the sorting logic in this file because the sorting queries rely on all of the same logic we use to filter by distance and I wanted to avoid duplicating a lot of this logic across multiple classes